Mathlib/NumberTheory/LSeries/Convergence.lean

```diff
@@ -85,27 +85,26 @@ lemma abscissaOfAbsConv_le_of_le_const_mul_rpow {f : ArithmeticFunction ℂ} {x
 open Filter in
 /-- If `‖f n‖` is `O(n^x)`, then the abscissa of absolute convergence
 of `f` is bounded by `x + 1`. -/
-lemma abscissaOfAbsConv_le_of_isBigO_rpow {f : ArithmeticFunction ℂ} {x : ℝ}
-    (h : f =O[atTop] fun n ↦ (n : ℝ) ^ x) : abscissaOfAbsConv f ≤ x + 1 := by
+lemma LSeries.abscissaOfAbsConv_le_of_isBigO_rpow {f : ℕ → ℂ} {x : ℝ}
+    (h : f =O[atTop] fun n ↦ (n : ℝ) ^ x) :
+    abscissaOfAbsConv f ≤ x + 1 := by
   rw [show x = x + 1 - 1 by ring] at h
   by_contra! H
   obtain ⟨y, hy₁, hy₂⟩ := EReal.exists_between_coe_real H
   exact (LSeriesSummable_of_isBigO_rpow (s := y) (EReal.coe_lt_coe_iff.mp hy₁) h
     |>.abscissaOfAbsConv_le.trans_lt hy₂).false
 
 /-- If `f` is bounded, then the abscissa of absolute convergence of `f` is bounded above by `1`. -/
-lemma abscissaOfAbsConv_le_of_le_const {f : ArithmeticFunction ℂ}
-    (h : ∃ C, ∀ n, ‖f n‖ ≤ C) : abscissaOfAbsConv f ≤ 1 := by
+lemma LSeries.abscissaOfAbsConv_le_of_le_const {f : ℕ → ℂ} (h : ∃ C, ∀ n ≠ 0, ‖f n‖ ≤ C) :
+    abscissaOfAbsConv f ≤ 1 := by
   convert abscissaOfAbsConv_le_of_le_const_mul_rpow (x := 0) ?_
   · norm_num
   · simpa only [norm_eq_abs, Real.rpow_zero, mul_one] using h
 
 open Filter in
 /-- If `f` is `O(1)`, then the abscissa of absolute convergence of `f` is bounded above by `1`. -/
-lemma abscissaOfAbsConv_le_one_of_isBigO_one {f : ArithmeticFunction ℂ}
-    (h : f =O[atTop] fun _ ↦ (1 : ℝ)) : abscissaOfAbsConv f ≤ 1 := by
+lemma LSeries.abscissaOfAbsConv_le_one_of_isBigO_one {f : ℕ → ℂ} (h : f =O[atTop] (1 : ℕ → ℝ)) :
+    abscissaOfAbsConv f ≤ 1 := by
```

Comment:
Not a comment on this PR, but a general comment about this proof: I've heard people express concern about proofs of the form `convert foo ; <complex tactics>` because they're susceptible to hard-to-fix breakage if definitions are refactored.

Reply:
I have replaced `norm_num` by a `simp only` call (which I hope does not count as "complex").